mylar/sabnzbd.py

```diff
@@ -112,55 +112,64 @@ def historycheck(self, nzbinfo):
-                            logger.warn('Either disable Completed Download Handling for SABnzbd within Mylar, or remove ComicRN from your category script in SABnzbd.')
-                            return {'status': 'double-pp', 'failed': False}
-
-                        if os.path.isfile(hq['storage']):
-                            logger.info('location found @ %s' % hq['storage'])
-                            found = {'status':   True,
-                                     'name':     ntpath.basename(hq['storage']), #os.pathre.sub('.nzb', '', hq['nzb_name']).strip(),
-                                     'location': os.path.abspath(os.path.join(hq['storage'], os.pardir)),
-                                     'failed':   False,
-                                     'issueid':  nzbinfo['issueid'],
-                                     'comicid':  nzbinfo['comicid'],
-                                     'apicall':  True,
-                                     'ddl':      False}
-                            break
-                        else:
-                            logger.info('no file found where it should be @ %s - is there another script that moves things after completion ?' % hq['storage'])
-                            return {'status': 'file not found', 'failed': False}
-
-                    elif hq['nzo_id'] == sendresponse and hq['status'] == 'Failed':
-                        #get the stage / error message and see what we can do
-                        stage = hq['stage_log']
-                        for x in stage[0]:
-                            if 'Failed' in x['actions'] and any([x['name'] == 'Unpack', x['name'] == 'Repair']):
-                                if 'moving' in x['actions']:
-                                    logger.warn('There was a failure in SABnzbd during the unpack/repair phase that caused a failure: %s' % x['actions'])
-                                else:
-                                    logger.warn('Failure occured during the Unpack/Repair phase of SABnzbd. This is probably a bad file: %s' % x['actions'])
-                                    if mylar.FAILED_DOWNLOAD_HANDLING is True:
-                                        found = {'status':   True,
-                                                 'name':     re.sub('.nzb', '', hq['nzb_name']).strip(),
-                                                 'location': os.path.abspath(os.path.join(hq['storage'], os.pardir)),
-                                                 'failed':   True,
-                                                 'issueid':  sendresponse['issueid'],
-                                                 'comicid':  sendresponse['comicid'],
-                                                 'apicall':  True,
-                                                 'ddl':      False}
-                                break
+
+        nzo_exists = False
+        for hq in histqueue['slots']:
```

Comment:
Not sure about the extra for/loop here as it seems kinda redundant, but maybe if we remove the 2nd for/loop since we already have the matching hq value, store that the hq value here (~line 119.. ```hqdata=hq```) and just run thru the subsequent if/elseif's referencing hqdata instead of hq, as there would be no for/loop.

Reply:
Yeah - good point.  let me refactor that a bit to remove the double loop.   That was just leftover from removing the while, and i hadn't paid too much attention to what was inside.